runtime/server/charts_api.go

```diff
@@ -119,3 +101,40 @@ func resolveChart(ctx context.Context, rt *runtime.Runtime, instanceID, name str
 
 	return spec, nil
 }
+
+func resolveAPI(ctx context.Context, rt *runtime.Runtime, instanceID, resolver string, resolverProps map[string]interface{}) (*runtimev1.API, error) {
+	var api *runtimev1.API
+	var err error
+
+	switch resolver {
+	case "Metrics":
+		resolverPropsPB, err := structpb.NewStruct(map[string]interface{}{
+			"sql": resolverProps["sql"],
+		})
+		if err != nil {
+			return nil, err
+		}
+		// TODO: are other fields needed?
+		api = &runtimev1.API{
+			Spec: &runtimev1.APISpec{
+				Resolver:           "Metrics",
+				ResolverProperties: resolverPropsPB,
+			},
+		}
+
+	case "API":
+		apiName, ok := resolverProps["api"].(string)
+		if !ok {
+			return nil, errors.New("api name is missing")
+		}
+		api, err = rt.APIForName(ctx, instanceID, apiName)
+		if err != nil {
+			return nil, err
+		}
+
+	default:
+		return nil, errors.New("unknown resolver")
+	}
```

Comment:
I would suggest refactoring `runtime.APIResolverOptions` to take `Resolver` and `ResolverProperties` as direct options instead of an `API`.

Since the point is to make the resolvers generic enough that all the code between the parser and the actual resolver doesn't need to know about the different available options.

Reply:
To make that work, you probably also need to add a new resolver called "API" that proxies to another resolver (and maybe checks to avoid infinite recursion if you want to be fancy?)